### PR TITLE
Fix take around synchronous edge cases

### PR DESCRIPTION
**Pull request by Andarist**

### CODE REVIEW

File: src/take.ts [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
 
     source(0, (t, d) => {
-      if (t === 0) talkback = d;
-      else if (t === 1) {
-        sink(1, d);
+      if (t === 0) {
+        talkback = d;
+        sink(0, (_: END) => {
+          ended = true;
+          talkback(_);
```
Comment (jvanbruegge) [inline src/take.ts]: Same here, we use `_` for explicitly ignored arguments. Just use `2` here

**Review by jvanbruegge — COMMENTED** [review]
Thanks

File: src/take.ts (hunk 1 above)
Comment (Andarist) [reply src/take.ts]: changed

**Comment by jvanbruegge** [comment]
Thanks

**Pull request merged**